tests/test_fourier_correlation.py

```diff
@@ -0,0 +1,278 @@
+import logging
+import os
+import tempfile
+
+import numpy as np
+import pytest
+
+from aspire.image import Image
+from aspire.noise import BlueNoiseAdder
+from aspire.numeric import fft
+from aspire.source import Simulation
+from aspire.utils import FourierRingCorrelation, FourierShellCorrelation, Rotation
+from aspire.volume import Volume
+
+from .test_utils import matplotlib_no_gui
+
+DATA_DIR = os.path.join(os.path.dirname(__file__), "saved_test_data")
+
+logger = logging.getLogger(__name__)
+
+IMG_SIZES = [
+    64,
+    65,
+]
+DTYPES = [
+    np.float64,
+    np.float32,
+]
+METHOD = ["fft", "nufft"]
+
+
+@pytest.fixture(params=DTYPES, ids=lambda x: f"dtype={x}")
+def dtype(request):
+    return request.param
+
+
+@pytest.fixture(params=METHOD, ids=lambda x: f"method={x}")
+def method(request):
+    return request.param
+
+
+@pytest.fixture(params=IMG_SIZES, ids=lambda x: f"img_size={x}")
+def img_size(request):
+    return request.param
+
+
+@pytest.fixture
+def image_fixture(img_size, dtype):
+    """
+    Serve up images with prescribed parameters.
+    """
+    # Load sample molecule volume
+    v = Volume(
+        np.load(os.path.join(DATA_DIR, "clean70SRibosome_vol.npy")), dtype=dtype
+    ).downsample(img_size)
+
+    # Instantiate ASPIRE's Rotation class with a set of angles.
+    thetas = [0, 0.123]
+    rots = Rotation.about_axis("z", thetas, dtype=dtype)
+
+    # Contruct the Simulation source.
+    src = Simulation(
+        L=img_size,
+        n=2,
+        vols=v,
+        offsets=0,
+        amplitudes=1,
+        C=1,
+        angles=rots.angles,
+        dtype=dtype,
+    )
+
+    img, img_rot = src.images[:]
+
+    noisy_src = Simulation(
+        L=img_size,
+        n=2,
+        vols=v,
+        offsets=0,
+        amplitudes=1,
+        C=1,
+        angles=rots.angles,
+        noise_adder=BlueNoiseAdder(var=np.var(img.asnumpy() * 0.5)),
+        dtype=dtype,
+    )
+    img_noisy = noisy_src.images[0]
```

Comment:
Could accomplish this by making just the noisy source and using `noisy_src.images` and `noisy_src.clean_images`.

Reply:
That is a good idea! Changed, thanks.